public/app/features/alerting/unified/components/GrafanaAlertmanagerDeliveryWarning.tsx

```diff
@@ -1,41 +1,70 @@
 import { css } from '@emotion/css';
+import { isEmpty } from 'lodash';
 import React from 'react';
 
 import { GrafanaTheme2 } from '@grafana/data/src';
 import { Alert, useStyles2 } from '@grafana/ui/src';
 
 import { AlertmanagerChoice } from '../../../../plugins/datasource/alertmanager/types';
+import { alertmanagerApi } from '../api/alertmanagerApi';
 import { GRAFANA_RULES_SOURCE_NAME } from '../utils/datasource';
 
 interface GrafanaAlertmanagerDeliveryWarningProps {
-  alertmanagerChoice?: AlertmanagerChoice;
   currentAlertmanager: string;
 }
 
-export function GrafanaAlertmanagerDeliveryWarning({
-  alertmanagerChoice,
-  currentAlertmanager,
-}: GrafanaAlertmanagerDeliveryWarningProps) {
+export function GrafanaAlertmanagerDeliveryWarning({ currentAlertmanager }: GrafanaAlertmanagerDeliveryWarningProps) {
   const styles = useStyles2(getStyles);
 
-  if (currentAlertmanager !== GRAFANA_RULES_SOURCE_NAME) {
+  const { useGetAlertmanagerChoiceQuery, useGetExternalAlertmanagersQuery } = alertmanagerApi;
+  const { currentData: alertmanagerChoice } = useGetAlertmanagerChoiceQuery();
+
+  const viewingInternalAM = currentAlertmanager === GRAFANA_RULES_SOURCE_NAME;
+
+  const interactsWithExternalAMs =
+    alertmanagerChoice && [AlertmanagerChoice.External, AlertmanagerChoice.All].includes(alertmanagerChoice);
+
+  const fetchExternalAMs = viewingInternalAM && interactsWithExternalAMs;
+
+  // only fetch external AMs when we have the right admin config
+  const { currentData: externalAMs } = useGetExternalAlertmanagersQuery(undefined, {
```

Comment:
There is one caveat there. The `getExternalAlertmanagers` endpoint requires `Admin` permissions, so the component would work only for Admins.
Maybe we could ask to BE folks to add information about the number of active external AMs to the `getAlertmanagerChoice` response. 
The `getExternalAlertmanagers` endpoint reveals AMs URLs, so there is probably no option to loosen its permission.

Reply:
Ah that's a pity – should we roll back the additional check for multiple external AMs?